File: smart-pos-core/src/main/java/com/interswitchng/smartpos/shared/services/iso8583/utils/FileUtils.java
package com.interswitchng.smartpos.shared.services.iso8583.utils;

import android.content.Context;
import android.content.res.AssetManager;

import java.io.IOException;
import java.io.InputStream;

public class FileUtils {

    public static String[] getFilesArrayFromAssets(Context context, String path) {
        AssetManager assetManager = context.getResources().getAssets();
        String[] files = null;
        try {
            files = assetManager.list(path);
        } catch (IOException e) {
            e.printStackTrace();
        }

        if(files!=null) {
            for (int i = 0; i < files.length; i++) {
                files[i] = path + "/" + files[i];
            }
        }

        return files;

    }

    public static byte[] getFromAssets(Context context) {
        InputStream in = null;
        try {
            in = context.getResources().getAssets().open("jpos.xml");
            //获取文件的字节数
            int lenght = in.available();
            //创建byte数组
            byte[] buffer = new byte[lenght];
            //将文件中的数据读到byte数组中
            in.read(buffer);
            return buffer;
        } catch (Exception e) {
            e.printStackTrace();

        } finally {
            if (in != null)
                try {
                    in.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
        }
        return null;
    }
}

File: smart-pos-core/src/main/java/com/interswitchng/smartpos/shared/services/iso8583/utils/FileUtils.kt
package com.interswitchng.smartpos.shared.services.iso8583.utils;

import android.content.Context;
import com.interswitchng.smartpos.R
import com.interswitchng.smartpos.shared.models.posconfig.TerminalConfig
import com.interswitchng.smartpos.shared.models.posconfig.EmvAIDs
import org.simpleframework.xml.core.Persister

import java.io.IOException;
import java.io.InputStream;

object FileUtils {

    internal fun <T> readXml(clazz: Class<T>, xmlFile: InputStream): T {
        val deserializer = Persister()
        return deserializer.read(clazz, xmlFile)
    }

    internal fun getFilesArrayFromAssets(context: Context, path: String): Array<String>? {
        val assetManager = context.resources.assets
        var files: Array<String>? = null
        try {
            files = assetManager.list(path);
        } catch (e: IOException) {
            e.printStackTrace()
        }

        if(files != null) {
            for (i in 0 until files.size) {
                files[i] = path + "/" + files[i]
            }
        }

        return files;
    }


    internal fun getFromAssets(context: Context): ByteArray? {
        var ins: InputStream? = null
        try {
            ins = context.resources.assets.open("jpos.xml");

            val length = ins.available()
            val buffer = ByteArray(length)
            ins.read(buffer)
            return buffer
        } catch (e: Exception) {
            e.printStackTrace();
        } finally {
            if (ins != null) {
                try {
                    ins.close()
                } catch (e: IOException) {
                    e.printStackTrace()
                }
            }
        }
        return null
    }

    internal fun getAids(xmlFile: InputStream): EmvAIDs {
        val deserializer = Persister()
        return deserializer.read(EmvAIDs::class.java, xmlFile)
    }

    internal fun getTerminalConfig(xmlFile: InputStream): TerminalConfig {
        val deserializer = Persister()
        return deserializer.read(TerminalConfig::class.java, xmlFile)
    }

    fun getConfigurations(context: Context): Pair<TerminalConfig, EmvAIDs> {
        // get resource streams
        val emvStr = context.resources.openRawResource(R.raw.isw_emv_config)
        val terminalStr = context.resources.openRawResource(R.raw.isw_terminal_config)
        // extract values
        val emvApps = getAids(emvStr)
        val terminalConfig = getTerminalConfig(terminalStr)

        return Pair(terminalConfig, emvApps)
    }
}
